### PR TITLE
[REVIEW] Fix getattr logic in GroupBy

### PR DESCRIPTION
**Pull request by shwina**
Noticed that our logic for `getattr` in GroupBy could lead to misleading error messages when an `AttributeError` is raised within any `property`s in GroupBy. See https://github.com/encode/django-rest-framework/issues/2108 for a description of the problem.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/tests/test_groupby.py [hunk 1]
```diff
@@ -1120,3 +1120,18 @@ def test_reset_index_after_empty_groupby():
         pdf.groupby("a").sum().reset_index(),
         gdf.groupby("a").sum().reset_index(),
     )
+
+
+def test_groupby_attribute_error():
+    err_msg = "Test error message"
+
+    class TestGroupBy(cudf.core.groupby.GroupBy):
```
Comment (shwina) [inline python/cudf/cudf/tests/test_groupby.py:1128]: We're depending on some internals of groupby in this test. I don't like it but couldn't figure out a better solution

Comment (kkraus14) [reply python/cudf/cudf/tests/test_groupby.py:1128]: By internals are you just referring to `cudf.core.groupby.GroupBy`?

**Comment by shwina** [comment]
rerun tests

**Pull request merged**